pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryConstructorRule.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.rule.codestyle;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.java.ast.ASTBlockStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceBody;
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTConstructorDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTEnumBody;
+import net.sourceforge.pmd.lang.java.ast.ASTExplicitConstructorInvocation;
+import net.sourceforge.pmd.lang.java.ast.ASTNameList;
+import net.sourceforge.pmd.lang.java.rule.AbstractIgnoredAnnotationRule;
+
+/**
+ * This rule detects when a constructor is not necessary;
+ * i.e., when there is only one constructor, it’s public, has an empty body,
+ * and takes no arguments.
+ */
+public class UnnecessaryConstructorRule extends AbstractIgnoredAnnotationRule {
+
+    @Override
+    protected Collection<String> defaultSuppressionAnnotations() {
+        Collection<String> defaultValues = new HashSet<>();
+        defaultValues.add("javax.inject.Inject");
+        return defaultValues;
+    }
+
+    @Override
+    public Object visit(ASTClassOrInterfaceBody node, Object data) {
+
+        if (isExplicitDefaultConstructor(node)
+            && hasDefaultAccessModifier((ASTClassOrInterfaceDeclaration) node.jjtGetParent(),
+            node.getFirstDescendantOfType(ASTConstructorDeclaration.class))) {
+            addViolation(data, node);
```

Comment:
The violation should probably be reported against the constructor directly, not the body or the class declaration